config/formats.js

```diff
@@ -420,65 +420,62 @@ let Formats = [
 		column: 2,
 	},
 	{
-		name: "[Gen 7] Suicide Cup",
-		desc: `Victory is obtained when all of your Pok&eacute;mon have fainted.`,
+		name: "[Gen 7] Metagamiate",
+		desc: `All Pok&eacute;mon gain an intrinsic -ate ability corresponding to their first type if not shiny, and second type (if applicable) if shiny.`,
 		threads: [
-			`&bullet; <a href="https://www.smogon.com/forums/threads/3633603/">Suicide Cup</a>`,
+			`&bullet; <a href="https://www.smogon.com/forums/threads/3604808/">Metagamiate</a>`,
 		],
 
-		mod: 'suicidecup',
-		forcedLevel: 100,
-		ruleset: ['Cancel Mod', 'Evasion Moves Clause', 'Endless Battle Clause', 'HP Percentage Mod', 'Moody Clause', 'Nickname Clause', 'Pokemon', 'Sleep Clause Mod', 'Species Clause', 'Team Preview'],
-		banlist: [
-			'Illegal', 'Unreleased', 'Shedinja', 'Infiltrator', 'Magic Guard', 'Misty Surge', 'Assault Vest', 'Choice Scarf', 'Explosion',
-			'Final Gambit', 'Healing Wish', 'Lunar Dance', 'Magic Room', 'Memento', 'Misty Terrain', 'Self-Destruct',
-		],
-		onValidateTeam: function (team) {
-			let problems = [];
-			if (team.length !== 6) problems.push(`Your team cannot have less than 6 Pok\u00e9mon.`);
-			let families = {};
-			for (const set of team) {
-				let pokemon = this.getTemplate(set.species);
-				if (pokemon.baseSpecies) pokemon = this.getTemplate(pokemon.baseSpecies);
-				if (pokemon.prevo) {
-					pokemon = this.getTemplate(pokemon.prevo);
-					if (pokemon.prevo) {
-						pokemon = this.getTemplate(pokemon.prevo);
-					}
-				}
-				if (!families[pokemon.species]) families[pokemon.species] = [];
-				families[pokemon.species].push(set.species);
+		mod: 'gen7',
+		ruleset: ['[Gen 7] OU'],
+		banlist: ['Dragonite', 'Kyurem-Black'],
+		onModifyMovePriority: -2,
+		onModifyMove: function (move, pokemon) {
+			if (move.type === 'Normal' && !move.isZ && !pokemon.hasAbility(['aerilate', 'galvanize', 'normalize', 'pixilate', 'refrigerate'])) {
+				let types = pokemon.getTypes();
+				let type = types.length < 2 || !pokemon.set.shiny ? types[0] : types[1];
+				move.type = type;
+				move.isMetagamiate = true;
 			}
-			for (const family in families) {
-				if (families[family].length > 1) problems.push(`${Chat.toListString(families[family])} are in the same evolutionary family.`);
+		},
+		onBasePowerPriority: 8,
+		onBasePower: function (basePower, attacker, defender, move) {
+			if (!move.isMetagamiate) return;
+			return this.chainModify([0x1333, 0x1000]);
+		},
+		validateSet: function (set, teamHas) {
```

Comment:
Just use onModifyTemplate for this..

Reply:
will it affect validation though?

Reply:
if you use `template = Object.assign({}, template);` with similar event looping code, then no